wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,220 @@
+
+		/*
+		 * Todo: change !== to === as this is only like this for development.
+		 * The WP_HTTPS_Detection doesn't work with my local SSL certificate.
+		 * Also, change $this->is_currently_https() to ! $this->is_currently_https().
+		 * This is also for development only.
+		 * This main if block should only run if the site can use HTTPS, but the 'home' and 'siteurl' options aren't HTTPS.
+		 */
+		if ( $this->is_currently_https() && true !== get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) ) :
+			?>
+			<p class="description">
+				<?php
+				echo wp_kses_post( sprintf(
+					/* Translators: %s: a link for more details */
+					esc_html__( 'HTTPS is essential to securing your WordPress site, we strongly suggest upgrading to HTTPS on your site. %s', 'pwa' ),
+					$https_more_details
+				) );
+				?>
+			</p>
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'HTTPS Upgrade', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( $upgrade_https_value, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_https_value ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description"><?php esc_html_e( 'Your site appears to support HTTPS', 'pwa' ); ?></p>
+
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'Upgrade Insecure URLs', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( $upgrade_insecure_content, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_insecure_content ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description">
+				<?php echo wp_kses_post( sprintf( $insecure_content_description, $insecure_content_more_details ) ); ?>
+			<p>
+			<?php
+		else :
+			/* Translators: %s: HTTPS more details link */
+			echo wp_kses_post( sprintf( __( 'Your site doesn\'t look to support HTTPS. %s', 'pwa' ), $https_more_details ) );
+		endif;
+	}
+
+	/**
+	 * Whether the options indicate that the site is currently using HTTPS.
+	 *
+	 * @return boolean
+	 */
+	public function is_currently_https() {
+		$urls = array( get_option( 'home' ), get_option( 'siteurl' ) );
+		foreach ( $urls as $url ) {
+			if ( 0 !== strpos( $url, self::HTTPS_PROTOCOL ) ) {
```

Comment:
Note that `strpos()` is going to search the entire string needlessly. If wanting to make this more performant (almost trivially) you could do `self::HTTPS_PROTOCOL !== substr( $url, 0, strlen( self::HTTPS_PROTOCOL ) )`. But that's hard to read. Maybe better to to use `wp_parse_url()` instead:

```php
'https' !== wp_parse_url( $url, PHP_URL_SCHEME )
```

Reply:
Thanks, [https://github.com/xwp/pwa-wp/pull/36/commits/f40eb5856cb134e343f670203bc1c4a87efd64a2](https://github.com/xwp/pwa-wp/pull/36/commits/f40eb5856cb134e343f670203bc1c4a87efd64a2) uses your snippet above.